polymer/src/amp-my-embed.html

```diff
@@ -0,0 +1,285 @@
+      }
+
+      urlChanged() {
+        this.fetchAndAttachAmpDoc_(this.fullUrl);
+      }
+
+      constructor() {
+        super();
+        // 'offline' is set to true if and when the document fetch fails.
+        this.state = {'offline': false};
+
+        /**
+         * `window.AMP` is set by the AMP runtime when it finishes loading.
+         * @const
+         * @private
+         */
+        this.ampReadyPromise_ = new Promise(resolve => {
+          (window.AMP = window.AMP || []).push(resolve);
+        });
+
+        /**
+         * XMLHTTPRequest that fetches the AMP document.
+         * @private
+         * @type {XMLHTTPRequest}
+         */
+        this.xhr_ = null;
+
+        /**
+         * Provides AMP functionality on the newly created shadow root after
+         * an AMP document is attached.
+         * @private
+         * @type {Object}
+         */
+        this.shadowAmp_ = null;
+
+        /**
+         * The root node of the shadow AMP.
+         * @note A single node must not be reused for multiple shadow AMP docs.
+         * @type {Element}
+         */
+        this.shadowRoot_ = null;
+
+        /** @private */
+        this.boundClickListener_ = this.clickListener_.bind(this);
+
+        /**
+         * Child element that will wrap the AMP shadow root.
+         * @private
+         * @type {Element}
+         */
+        this.container_ = document.getElementById('container');
+
+      }
+
+      ready() {
+        super.ready();
+        this.$.name.textContent = this.shadowAmp_;
+      }
+
+  /**
+   * Fetches the AMP document at `url` and attaches it as a shadow root.
+   * @private
+   * @param {string} url
+   */
+  fetchAndAttachAmpDoc_(url) {
+    this.fetchDocument_(url).then(doc => {
+      return this.ampReadyPromise_.then(amp => {
+        // Hide navigational and other unwanted elements before displaying.
+        this.hideUnwantedElementsOnDocument_(doc);
+
+        // Replace the old shadow root with a new div element.
+        const oldShadowRoot = this.shadowRoot_;
+        this.shadowRoot_ = document.createElement('div');
+        if (oldShadowRoot) {
+          this.$.name.replaceChild(this.shadowRoot_, oldShadowRoot);
+        } else {
+          this.$.name.appendChild(this.shadowRoot_);
+        }
+
+        // Attach the shadow document to the new shadow root.
+        this.shadowAmp_ = amp.attachShadowDoc(this.shadowRoot_, doc, url);
+      });
+    }).catch(error => {
+      this.setState({'offline': true});
+    });
+  }
+
+  /**
+   * Cleans up internal state of current shadow AMP document.
+   * @private
+   */
+  closeShadowAmpDoc_() {
+    if (typeof this.shadowAmp_.close === 'function') {
+      this.shadowAmp_.close();
+    }
+  }
+
+  /**
+   * Hides elements (e.g. banners) that would clash with the app shell.
+   * @param {!Document} doc
+   * @private
+   */
+   hideUnwantedElementsOnDocument_(doc) {
+     const banners = doc.getElementsByClassName('banner');
```

Comment:
In general, webcomponents should not rely on any state outside of the element. E.g. this element depends on state that is in the document root instead. Is that intended?

Reply:
would it be better to move this function in the sampledetail.html? what are the alternatives to achieve the same goal?

Reply:
I usually move the elements I depend on the `light-dom`. I am not sure if you can do that in this case, but it would be something like

```html
<amp-my-embed><div class='banner'></div><amp-my-embed>
```
I am not sure if you can do it in this case, as I see no other reference to `banner` in this PR. Is this something AMP handles?

Reply:
Let's remove this for now as you are right I haven't a banner for now or other navigation.